keyboards/xd87/xd87.h

```diff
@@ -42,14 +42,14 @@
     K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B,      K3D,                     \
     K40,      K42, K43, K44, K45, K46, K47, K48, K49, K4A, K4B, K4C,                K4F,      \
     K50, K51, K52,                K58,                K5A, K5B, K5C, K5D,      K5E, K5F, K5G  \
-) LAYOUT_all( \
-    K00,   KC_NO, K02,   K03,   K04,   K05,   K06,   K07,   K08,   K09,   K0A,   K0B,   K0C,   K0D,          K0E,   K0F,   K0G,   \
-    K10,   K11,   K12,   K13,   K14,   K15,   K16,   K17,   K18,   K19,   K1A,   K1B,   K1C,   K1D, KC_NO,   K1E,   K1F,   K1G,   \
-    K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   K2C,   K2D,          K2E,   K2F,   K2G,   \
-    K30,   K31,   K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   K3B,   KC_NO, K3D,                               \
-    K40,   KC_NO, K42,   K43,   K44,   K45,   K46,   K47,   K48,   K49,   K4A,   K4B,   K4C,   KC_NO,        KC_NO, K4F,   KC_NO, \
-    K50,   K51,   K52,                 KC_NO,        K58,                 K5A,   K5B,   K5C,   K5D,          K5E,   K5F,   K5G    \
-)
+) { \
+    {K00,   KC_NO, K02,   K03,   K04,   K05,   K06,   K07,   K08,   K09,   K0A,   K0B,   K0C,   K0D,          K0E,   K0F,   K0G}, \
+    {K10,   K11,   K12,   K13,   K14,   K15,   K16,   K17,   K18,   K19,   K1A,   K1B,   K1C,   K1D,          K1E,   K1F,   K1G}, \
+    {K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   K2C,   K2D,          K2E,   K2F,   K2G}, \
+    {K30,   K31,   K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   K3B,   KC_NO, K3D}, \
```

Comment:
Are you sure this works correctly? Each row should have `MATRIX_COLS` number of elements, with nonexistent matrix positions blanked out by `KC_NO`s.
Weirdly, Travis doesn't seem to mind, but perhaps GCC is just silently padding the truncated rows out.

Reply:
> perhaps GCC is just silently padding the truncated rows out.

That's exactly what happens – GCC pads the truncated rows out with NULL, which ends up the same as using `KC_NO` (which is defined as `0x00` in `tmk_core/common/keycode.h`. This paradigm used to be all over the repo when I was starting out with contributing to QMK.

Reply:
Back to add: As for the parts that skip entries in the middle of a row, I'm not convinced that works.

Reply:
The funny part is, that the bottom row is broken. I'm fixing it now.